### PR TITLE
Added unit tests

### PR DESCRIPTION
**Pull request by hamersaw**
Fleshed out some unit testing. Purposefully left out passthrough, it seems a unit test here is just writing the opposite and doesn't really test anything.

### CODE REVIEW

File: pkg/controller/crdoffloadstore/active.go [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+		return err
+	}
+
+	a.metrics.TotalItems.Dec()
+	delete(a.store, dataReference.String())
+
+	return nil
+}
+
+func NewActiveCRDOffloadStore(crdOffloadStore CRDOffloadStore, scope promutils.Scope) CRDOffloadStore {
```
Comment (ckiosidis) [inline pkg/controller/crdoffloadstore/active.go:60]: this is a rename, right?

File: pkg/controller/crdoffloadstore/lru.go [hunk 2]
```diff
@@ -35,19 +35,21 @@ func (l *lruCRDOffloadStore) Get(ctx context.Context, dataReference v1alpha1.Dat
 			l.metrics.CacheReadError.Inc()
 			return nil, fmt.Errorf("cached item in crd offload store is not expected type '*v1alpha1.StaticWorkflowData'")
 		}
+
 		l.metrics.CacheHit.Inc()
 		return staticWorkflowData, nil
 	}
 
-	timer := l.metrics.FetchLatency.Start()
+	l.metrics.CacheMiss.Inc()
```
Comment (ckiosidis) [inline pkg/controller/crdoffloadstore/lru.go:43]: 👍

File: pkg/controller/crdoffloadstore/active.go (hunk 1 above)
Comment (hamersaw) [reply pkg/controller/crdoffloadstore/active.go:60]: Correct. I think that the "Inmemory" in terms of a CRD offload store implies that there is no backing store. Technically the LRU cache is also "in memory". So maybe "active" means that we are caching all of the workflows that are currently active. I'm not tied to the naming, OK changing if you feel stronlgy.

File: pkg/controller/handler_test.go [hunk 3]
```diff
@@ -870,4 +873,15 @@ func TestPropellerHandler_OffloadedCrd(t *testing.T) {
...
+
+	t.Run("Error", func(t *testing.T) {
+		scope := promutils.NewTestScope()
+
+		offloadmock := &crdoffloadstoremock.CRDOffloadStore{}
+		p := NewPropellerHandler(ctx, cfg, s, offloadmock, exec, scope)
+
+		offloadmock.OnGetMatch(mock.Anything, mock.Anything).Return(nil, fmt.Errorf("foo"))
+		err := p.Handle(ctx, namespace, name)
+		assert.Error(t, err)
```
Comment (ckiosidis) [inline pkg/controller/handler_test.go:885]: 👍

**Review by ckiosidis — APPROVED** [review]

**Pull request merged**